### PR TITLE
[plugin-mobile-app] Make JavaScript executing steps available for mob…

### PR DESCRIPTION
**Pull request by ikalinin1**
…ile app tests

### CODE REVIEW

File: vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java [hunk 1]
```diff
@@ -88,10 +74,10 @@
     private static final String VARIABLE_NAME = "variableName";
 
     @Mock
-    private ISoftAssert softAssert;
+    private SoftAssert softAssert;
```
Comment (valfirst) [inline vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java:77]: what is the reason of this change?

File: vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java [hunk 2]
```diff
@@ -88,10 +74,10 @@
...
 
     @Mock
-    private IBaseValidations mockedBaseValidations;
+    private BaseValidations mockedBaseValidations;
```
Comment (valfirst) [inline vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java:80]: what is the reason of this change?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java (hunk 1 above)
Comment (ikalinin1) [reply vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java:77]: Follow common practice

Comment (ikalinin1) [reply vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java:77]: I changed the interfaces in step classes and obviously Mockito unable to inject interface

Comment (valfirst) [reply vividus-plugin-web-app/src/test/java/org/vividus/steps/ui/web/CodeStepsTests.java:77]: https://github.com/vividus-framework/vividus/pull/3139

File: docs/modules/plugins/partials/generic-ui-steps.adoc [hunk 3]
```diff
@@ -405,3 +405,51 @@ When I find >= `0` elements `xpath(//*[@class='menu enabled'])` and while they e
...
+----
+When I execute javascript `document.querySelector('#assistant').remove()` with arguments:
+----
+
+.Inject image
+[source,gherkin]
+----
+When I execute javascript `sauce:inject-image` with arguments:
+|value                                                                                       |type  |
+|iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNk+A8AAQUBAScY42YAAAAASUVORK5CYII=|string|
```
Comment (uarlouski) [inline docs/modules/plugins/partials/generic-ui-steps.adoc:434]: please add some example with object type to make it clear that JSON object is expected

File: vividus-tests/src/main/resources/story/integration/CodeSteps.story [hunk 4]
```diff
@@ -10,11 +10,19 @@ Given I am on a page with the URL '${vividus-test-site-url}'
...
+
 Scenario: Step verification: "When I execute async javascript `$jsCode` and save result to $scopes variable `$variableName`"
 When I execute async javascript `arguments[0](JSON.stringify(window.performance.timing))` and save result to scenario variable `timings`
 Then number of JSON elements from `${timings}` by JSON path `$.connectStart` is = 1
+
+
+Scenario: Step verification: "When I execute javascript `$jsCode` with arguments:$args"
+Then number of elements found by `xpath(//img)` is = `1`
+When I execute javascript `document.querySelector('[name="vividus-logo"]').remove()` with arguments:
+Then number of elements found by `xpath(//img)` is = `0`
```
Comment (uarlouski) [inline vividus-tests/src/main/resources/story/integration/CodeSteps.story:28]: should we refresh the page at the end to return it to its initial state? the change is extremely low but we might accidentally add one more scenario after this one that will rely on the logo element and it will fail because we didn't notice logo remove operation in previous scenario

Comment (ikalinin1) [reply vividus-tests/src/main/resources/story/integration/CodeSteps.story:28]: If we add an extra scenario then there should be refresh, now its redundant

File: vividus-tests/src/main/resources/story/system/mobile_app/MobileAppStepsTests.story [hunk 5]
```diff
@@ -1,4 +1,4 @@
-Description: Integration tests for mobile app steps
+]Description: Integration tests for mobile app steps
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/system/mobile_app/MobileAppStepsTests.story:1]:
```suggestion
Description: Integration tests for mobile app steps
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]
